src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -302,15 +304,36 @@ private boolean canSettle(Authentication a) {
             return true;
         }
         final Set<String> submitters = Sets.newHashSet(submitter.split(","));
-        if (submitters.contains(a.getName()))
+        if (isMemberOf(a.getName(), submitters))
             return true;
         for (GrantedAuthority ga : a.getAuthorities()) {
-            if (submitters.contains(ga.getAuthority()))
+            if (isMemberOf(ga.getAuthority(), submitters))
                 return true;
         }
         return false;
     }
 
+    /**
+     * Checks if the provided userId is contained in the submitters list, using {@link SecurityRealm#getUserIdStrategy()} comparison algorithm.
+     * Main goal is to respect here the case sensitivity settings of the current security realm
+     * (which default behavior is case insensitivity).
+     *
+     * @param userId the id of the user if it is matching one of the submitters using {@link IdStrategy#equals(String, String)}
+     * @param submitters the list of authorized submitters
+     * @return true is userId was found in submitters, false if not.
+     *
+     * @see {@link jenkins.model.IdStrategy#CASE_INSENSITIVE}.
+     */
+    private boolean isMemberOf(String userId, Set<String> submitters) {
+        final IdStrategy userIdStrategy = Jenkins.getActiveInstance().getSecurityRealm().getUserIdStrategy();
```

Comment:
This should be using `getGroupIdStrategy` in the second case. So, factor out an `IdStrategy` method parameter.

Reply:
Addressed in https://github.com/jenkinsci/pipeline-input-step-plugin/pull/28